crates/sui-config/src/genesis.rs

```diff
@@ -698,18 +693,27 @@ impl Builder {
 
         // Verify that all the validators were properly created onchain
         let system_object = genesis.sui_system_object();
-        assert_eq!(system_object.epoch, 0);
+        assert_eq!(system_object.epoch(), 0);
 
         for (validator, onchain_validator) in validators
             .iter()
             .map(|genesis_info| &genesis_info.info)
-            .zip(system_object.validators.active_validators.iter())
+            .zip(
+                system_object
+                    .into_sui_system_state_summary()
+                    .active_validators,
+            )
         {
-            let metadata = onchain_validator.verified_metadata();
-            assert_eq!(validator.sui_address(), metadata.sui_address);
-            assert_eq!(validator.protocol_key(), metadata.sui_pubkey_bytes());
-            assert_eq!(validator.name(), &metadata.name);
-            assert_eq!(validator.network_address(), &metadata.net_address);
+            assert_eq!(validator.sui_address(), onchain_validator.sui_address);
+            assert_eq!(
+                validator.protocol_key().as_bytes(),
+                &onchain_validator.protocol_pubkey_bytes
+            );
+            assert_eq!(validator.name(), &onchain_validator.name);
+            assert_eq!(
+                validator.network_address(),
+                &Multiaddr::try_from(onchain_validator.net_address.clone()).unwrap()
+            );
```

Comment:
So Genesis tooling will 100% need to deeply inspect the inner types and i'm currently working on this as we speak (this is for auditing purposes)

Reply:
That's ok. Here we could also do into_genesis_version. The only thing I am trying to avoid is to depend on this type in sui-node startup path

Reply:
Reverted this part.

Reply:
completely agree with that goal (eliminating this in the sui-node startup path).

So to that end feel free to make these methods private if need be.